portal-api/api-tests/v1/utilisation-report-service/last-uploaded.api-test.js

```diff
@@ -36,47 +30,36 @@ describe('GET /v1/banks/:bankId/utilisation-reports/last-uploaded', () => {
         month: lastUploadedReportPeriodMonth,
         year,
       },
-    },
-    dateUploaded: lastUploadedReportDateUploaded,
-  };
+    })
+    .withDateUploaded(lastUploadedReportDateUploaded)
+    .build();
 
   beforeAll(async () => {
-    await databaseHelper.wipe([DB_COLLECTIONS.UTILISATION_REPORTS]);
+    await wipeAllUtilisationReports();
 
     testUsers = await testUserCache.initialise(app);
     aPaymentReportOfficer = testUsers().withRole(PAYMENT_REPORT_OFFICER).one();
     matchingBankId = aPaymentReportOfficer.bank.id;
 
     const { bank } = aPaymentReportOfficer;
-    lastUploadedReport.bank = bank;
-
-    mockUtilisationReports = [
-      lastUploadedReport,
-      {
-        ...MOCK_NOT_RECEIVED_REPORT_WITHOUT_ID,
-        bank,
-        reportPeriod: {
-          start: {
-            month: lastUploadedReportPeriodMonth + 1,
-            year,
-          },
-          end: {
-            month: lastUploadedReportPeriodMonth + 1,
-            year,
-          },
+    lastUploadedReport.bankId = bank.id;
+    const notReceivedReport = UtilisationReportEntityMockBuilder.forStatus(UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED)
+      .withId(6)
```

Comment:
Is this just us needing to add an id? would we want to just let the DB generate this or are we trying to override so that we can refer to it later?

Reply:
So, the mock builder adds the id as 1, so if i am writing a test with more than one entity, I need to set the ids for each entity to different values, otherwise when I save the entities after the first one it doesn't add a second entity but it also didn't throw as big an error as I would have liked it to 😅